caddy.go

```diff
@@ -160,6 +163,8 @@ func (i *Instance) Restart(newCaddyfile Input) (*Instance, error) {
 	}
 	instancesMu.Unlock()
 
+	i.Stop()
```

Comment:
Turns out `i.Stop()` does the same thing as the for loop starting on line 155/158 does (removing it from the list).